sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -140,3 +141,49 @@ case class Sha2(left: Expression, right: Expression)
     """
   }
 }
+
+/**
+ * A function that calculates a sha1 hash value and returns it as a hex string
+ * For input of type [[BinaryType]] or [[StringType]]
+ */
+case class Sha1(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+
+  override def dataType: DataType = StringType
+
+  override def expectedChildTypes: Seq[DataType] = Seq(BinaryType, StringType)
```

Comment:
This should be `Seq(BinaryType)`, you can checkout out  `ExpectedInputConversion` to see how `ExpectsInputTypes` works, take Sha2 as example.